lib/compat/wordpress-6.0/class-wp-theme-json-gutenberg.php

```diff
@@ -529,6 +560,143 @@ protected static function get_metadata_boolean( $data, $path, $default = false )
+	 * )
+	 * ```
+	 *
+	 * @param array   $styles Styles to process.
+	 * @param array   $settings Theme settings.
+	 * @param array   $properties Properties metadata.
+	 * @param string  $selector Selector for styles.
+	 * @param boolean $use_root_vars Whether to use root variables.
+	 * @return array Returns the modified $declarations.
+	 */
+	protected static function compute_style_properties( $styles, $settings = array(), $properties = null, $selector = null, $use_root_vars = null ) {
+		if ( null === $properties ) {
+			$properties = static::PROPERTIES_METADATA;
+		}
+
+		$declarations             = array();
+		$root_variable_duplicates = array();
+
+		if ( empty( $styles ) ) {
+			return $declarations;
+		}
+
+		foreach ( $properties as $css_property => $value_path ) {
+			$value = static::get_property_value( $styles, $value_path );
+
+			if ( strpos( $css_property, '--wp--style--root--' ) === 0 && static::ROOT_BLOCK_SELECTOR !== $selector ) {
+				continue;
+			}
+
+			if ( strpos( $css_property, '--wp--style--root--' ) === 0 && $use_root_vars ) {
+				$root_variable_duplicates[] = substr( $css_property, strlen( '--wp--style--root--' ) );
+			}
+
+			// Root padding requires special logic to split shorthand values.
+			if ( '--wp--style--root--padding' === $css_property && is_string( $value ) ) {
+
+				$shorthand_top    = '0';
+				$shorthand_right  = '0';
+				$shorthand_bottom = '0';
+				$shorthand_left   = '0';
+
+				$separate_values = explode( ' ', $value );
+
+				switch ( count( $separate_values ) ) {
+					case 1:
+						$shorthand_top    = $separate_values[0];
+						$shorthand_right  = $separate_values[0];
+						$shorthand_bottom = $separate_values[0];
+						$shorthand_left   = $separate_values[0];
+						break;
+					case 2:
+						$shorthand_top    = $separate_values[0];
+						$shorthand_right  = $separate_values[1];
+						$shorthand_bottom = $separate_values[0];
+						$shorthand_left   = $separate_values[1];
+						break;
+					case 3:
+						$shorthand_top    = $separate_values[0];
+						$shorthand_right  = $separate_values[1];
+						$shorthand_bottom = $separate_values[2];
+						$shorthand_left   = $separate_values[1];
+						break;
+					case 4:
+						$shorthand_top    = $separate_values[0];
+						$shorthand_right  = $separate_values[1];
+						$shorthand_bottom = $separate_values[2];
+						$shorthand_left   = $separate_values[3];
+						break;
+				}
+
+				$all_properties = array(
+					array(
+						'name'  => '--wp--style--root--padding-top',
+						'value' => $shorthand_top,
+					),
+					array(
+						'name'  => '--wp--style--root--padding-right',
+						'value' => $shorthand_right,
+					),
+					array(
+						'name'  => '--wp--style--root--padding-bottom',
+						'value' => $shorthand_bottom,
+					),
+					array(
+						'name'  => '--wp--style--root--padding-left',
+						'value' => $shorthand_left,
+					),
+				);
+
+				$declarations = array_merge( $declarations, $all_properties );
+
```

Comment:
I don't know if it's been brought up yet, but this is a lot of bespoke code for root padding values future iterations. I'm not sure where we'd put it in future iterations. 

I'm not saying it's a blocker, just curious to learn whether we need a separate processing method for root block selectors to keep these methods uncluttered.

I'd speculate that root style CSS vars could be useful elsewhere, e.g., `--wp--style--root--block-gap`

Reply:
Yeah I agree a separate method for root selectors might make this tidier.

Reply:
Done!